packages/manager/modules/adp/src/service/cluster/cluster.controller.js

```diff
@@ -1,28 +1,166 @@
+import find from 'lodash/find';
+import filter from 'lodash/filter';
+import map from 'lodash/map';
+import uniq from 'lodash/uniq';
+import forEach from 'lodash/forEach';
+
 export default class {
   /* @ngInject */
-  constructor() {
+  constructor($stateParams, ADP_COMPUTE, ADP_STATUS_MAP,
+    adpService, CucControllerHelper, CucServiceHelper) {
+    this.adpService = adpService;
+    this.cucControllerHelper = CucControllerHelper;
+    this.cucServiceHelper = CucServiceHelper;
+    this.serviceName = $stateParams.serviceName;
+    this.totalCores = 0;
+    this.totalRam = 0;
+    this.totalStorage = 0;
+    this.ALL_NODES = 'ALL';
+
     // MOCK DATA : Will be removed after integrating with API
-    this.filters = [
-      'Role',
+    this.nodeFilters = [
+      'Node type',
       'Region',
     ];
-    this.regions = [
-      'Beauharnois(BHS3)',
-    ];
-    this.roles = [
-      'MASTER',
-      'UTILITY',
-      'BASTION',
-      'WORKER',
-      'EDGE',
-    ];
+
+    this.regions = [];
+
+    this.nodeTypeList = [];
+
+    this.ADP_STATUS_MAP = ADP_STATUS_MAP;
+    this.ADP_COMPUTE = ADP_COMPUTE;
     this.filterBy = null;
     this.selectedRegion = null;
     this.selectedRole = null;
   }
 
   selectFilter() {
-    this.selectedRole = null;
+    this.selectedNodeType = null;
     this.selectedRegion = null;
+    this.clusterList = this.clusterNodes.data.nodes;
+  }
+
+  $onInit() {
+    this.populateList();
+  }
+
+  populateList() {
+    this.fetchAdpDetails()
+      .then((platform) => {
+        this.flavorPromise = this.fetchFlavors(platform.project_id, platform.os_region);
+      });
+    this.getClusterNodes()
+      .then(({ nodes }) => {
+        this.totalStorage = 0;
+        forEach(nodes, (node) => {
+          this.totalStorage += node.storage;
+        });
+        this.clusterList = nodes;
+
+        this.nodeTypeList = uniq(map(this.clusterList, 'node_type'));
+        this.regions = (uniq(map(this.clusterList, 'region')));
+        this.nodeTypeList.unshift(this.ALL_NODES);
+        this.regions.unshift(this.ALL_NODES);
+      })
+      .then(() => this.fetchPublicCloudDetails());
+  }
+
+  /**
+   * fetch all cluster nodes
+   *
+   * @returns array of cluster nodes
+   */
+  getClusterNodes() {
+    this.clusterNodes = this.cucControllerHelper.request.getHashLoader({
+      loaderFunction: () => this.adpService.getClusterNodes(this.serviceName)
+        .catch(error => this.cucServiceHelper.errorHandler('adp_get_nodes_error')(error)),
+    });
+    return this.clusterNodes.load();
+  }
+
+  /**
+   * Loads flavor details like ram, cores and storage
+   */
+  loadRowWithFlavorDetails(row) {
+    return this.flavorPromise.then((flavors) => {
+      const flavor = find(flavors, { name: row.flavor_type });
+      this.totalCores += flavor.vcpus;
```

Comment:
use _.sumBy

Reply:
This cant be done as it is a global sum and will be added for every row.